### ISSUE TITLE
KeyboardAvoidingView not working with react-nativgation (and iOS)

### ISSUE THREAD

**Issue by tomwaitforitmy** [react-navigation/react-navigation#11244]
### Current behavior

[You can check my repro snack.](https://snack.expo.dev/@tomwaitforitmy/keyboardavoidingview)

Like [last time](https://github.com/react-navigation/react-navigation/issues/10583), `KeyboardAvoidingView` is broken with `react-navigation`. This time, however, both bottom tab navigators don't work: `@react-navigation/bottom-tabs` and `@react-navigation/material-bottom-tabs`. This time `react-native-paper` can be excluded, because it is not used in  `@react-navigation/bottom-tabs`. 

![grafik](https://user-images.githubusercontent.com/25305424/221355808-cd0b8810-4c52-4a83-8091-2df3867ff9be.png)

Side note: I am not sure if this is an iOS only bug or not. `KeyboardAvoidingView` is not working on Android either, however, on Android the Keyboard does not hide `<TextInput>`. You do not have to use `KeyboardAvoidingView` at all. On Android, the keyboard works out of the box. On iOS, you need `KeyboardAvoidingView` which breaks once you use stack navigators together with tab navigators.

You think my snack is huge or has flaws?
Please try these simple steps to reproduce:
1. `npx expo init <yourReproNameHere>`
2. Use the "tabs" template
3. Fill any of the two screens with anything (I used `<Button/>`)
4. Add a `<TextInput>` at the end of the screen
5. The keyboard hides the `<TextInput>` and `<KeyboardAvoidingView>` will not fix it

Optionally:
* Add a `<Scrollview>`
* Use `@react-navigation/material-bottom-tabs` which is not working either
* `import { useHeaderHeight } from "@react-navigation/elements";` is not fixing anything either. [You can check my repro snack.](https://snack.expo.dev/@tomwaitforitmy/keyboardavoidingview)

### Expected behavior

The keyboard should not hide the `<TextInput/>`

### Reproduction

https://snack.expo.dev/@tomwaitforitmy/keyboardavoidingview

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [X] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.0.2
| @react-navigation/bottom-tabs          | 6.0.5
| @react-navigation/material-top-tabs    | 6.2.14
| @react-navigation/native-stack         | 6.1.0 & 6.6.1
| @react-navigation/elements | 1.3.1
| react-native-safe-area-context         | 4.4.1
| react-native-screens                   | 3.18.0
| react-native                           | 0.70.5
| react-native-paper | 5.2.0
| expo                                   | 47.0.12

**Comment by tomwaitforitmy** [comment]
> Couldn't find version numbers for the following packages in the issue:
> 
>     * `@react-navigation/stack`
> 
> 
> Can you update the issue to include version numbers for those packages? The version numbers must match the format 1.2.3.
> 
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
>     * `@react-navigation/native` (found: `6.0.2`, latest: `6.1.5`)
> 
>     * `@react-navigation/bottom-tabs` (found: `6.0.5`, latest: `6.5.6`)
> 
>     * `@react-navigation/material-top-tabs` (found: `6.2.14`, latest: `6.6.1`)
> 
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

I checked with `@react-navigation/native` `6.1.5` and `@react-navigation/bottom-tabs` `6.5.6`. Still repro.

**Comment by tomwaitforitmy** [comment]
bump

**Comment by tomwaitforitmy** [comment]
Still bump 😏

**Comment by snowxcone** [comment]
Hello! in your expo, I got it to work by changing your MyKeyboardAvoidingView.js file to the following
`import { useHeaderHeight } from "@react-navigation/elements";
import React from "react";
import { KeyboardAvoidingView, StyleSheet, Platform } from "react-native";

const MyKeyboardAvoidingView = ({ children, headerHeight }) => {
  //as suggested in many places online adjust the vertical offset with the header height.
  const myHeaderHeight = useHeaderHeight();

  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior={Platform.OS === "ios" ? "padding" : "height"}
      //if you add 64 here, it works quite nice for iOS, but after you close the keyboard,
      //the view is messed up by 64(?) pixels
      keyboardVerticalOffset={myHeaderHeight+47}
    >
      {children}
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 5,
    width: "100%",
  },
});

export default MyKeyboardAvoidingView;`

Let me know if this works for you! :)

**Comment by rifad4u** [comment]
Hi @tomwaitforitmy , Pls wrap the keyboard avoiding view in the MainTabNavigator instead of wrapping it in the dev screen.
It should be like this 

function MainTabNavigator() {
  return (
    <MyKeyboardAvoidingView>
      <Tab.Navigator>
        <Tab.Screen name="Dev" component={DevStackContainer} />
      </Tab.Navigator>
    </MyKeyboardAvoidingView>
  );
}

**Comment by tomwaitforitmy** [comment]
@rifad4u you are right. That works. Not very intuitive to wrap the tab navigator with `KeyboardAvoidingView`. My assumption would be to wrap components with `TextInput` in contrast to basically the whole app once. That behavior should be mentioned in the docs. I will try to add it. If I get rejected, it should not be the desired behavior.

**Comment by tomwaitforitmy** [comment]
@snowxcone The screenshot shows what I get from your magic number solution. I don't they ever work on all devices. There is never a decent explanation for the +47 and the +64 anyways. These values might work for your iPhone, but for mine they don't.
![grafik](https://user-images.githubusercontent.com/25305424/230776597-4d0139bc-d6d6-4a6e-9811-85c2cbb27568.png)

**Comment by tomwaitforitmy** [comment]
@rifad4u I created a PR here: https://github.com/react-navigation/react-navigation.github.io/pull/1229
Please add extra background information if you have any. I am still not 100% sure why I should use `KeyboardAvoidingView` like you proposed. I only see that it is working.
